sdk/core/azure-core/src/io/body_stream.cpp

```diff
@@ -190,3 +190,43 @@ size_t FileBodyStream::OnRead(uint8_t* buffer, size_t count, Azure::Core::Contex
 void FileBodyStream::Rewind() { m_randomAccessFileBodyStream->Rewind(); }
 
 int64_t FileBodyStream::Length() const { return m_randomAccessFileBodyStream->Length(); }
+
+ProgressBodyStream::ProgressBodyStream(
+    BodyStream* bodyStream,
+    std::function<void(int64_t bytesTransferred)> callback)
+    : m_bodyStream(bodyStream), m_bytesTransferred(0), m_callback(std::move(callback))
+{
+  if (bodyStream == nullptr)
+  {
+    throw std::runtime_error("Parameter 'bodystream' cannot be null");
+  }
+
+  m_callback(m_bytesTransferred);
+}
+
+ProgressBodyStream::~ProgressBodyStream()
+{
+  m_bodyStream = nullptr;
+  m_callback = nullptr;
```

Comment:
This is kind of new to me. Why do we need to do this?  Wouldn't the default dctor call each member dctor?

Reply:
We don't need this. We don't need to define such a destructor at all.

Reply:
I agree, we don't need this